### PR TITLE
.github/workflows: skip net/dnsfallback go generate check on release branch

### PR DESCRIPTION
**Pull request by josharian**

### CODE REVIEW

File: .github/workflows/go_generate.yml [hunk 1]
```diff
@@ -28,7 +28,14 @@ jobs:
         run: |
           mkdir gentools
           go build -o gentools/stringer golang.org/x/tools/cmd/stringer
-          PATH="$PATH:$(pwd)/gentools" go generate ./...
+          if [[ "${{github.ref}}" == release-branch/* ]]
```
Comment (bradfitz) [inline .github/workflows/go_generate.yml:31]: whoa, == matches against globs? TIL.

File: .github/workflows/go_generate.yml [hunk 2]
```diff
@@ -28,7 +28,14 @@ jobs:
         run: |
           mkdir gentools
           go build -o gentools/stringer golang.org/x/tools/cmd/stringer
```
Comment (bradfitz) [inline .github/workflows/go_generate.yml:30]: I don't think we need this anymore. Awhile back I updated all the raw "stringer" calls to use "go run golang.org/x/tools/cmd/stringer" instead.

**Review by bradfitz — APPROVED** [review]

File: .github/workflows/go_generate.yml (hunk 1 above)
Comment (josharian) [reply .github/workflows/go_generate.yml:31]: Shell. 🤪

File: .github/workflows/go_generate.yml (hunk 2 above)
Comment (josharian) [reply .github/workflows/go_generate.yml:30]: Ack. Will fix in separate commit in this PR.

**Review by bradfitz — APPROVED** [review]

**Comment by josharian** [comment]
Windows race failure is unrelated. Filed https://github.com/tailscale/tailscale/issues/2855 for later. (I'm busy working on a different flaky test now. 😢)

**Pull request merged**